src/praisonai-agents/praisonaiagents/telemetry/performance_utils.py

```
@@ -0,0 +1,576 @@
+                    indent = "  " * len(call_stack)
+                    status = "✅" if success else "❌"
+                    lines.append(f"{indent}{status} END   {func_name} ({duration:.3f}s)")
+        
+        return "\n".join(lines)
+    
+    def _create_mermaid_diagram(self, flow_data: List[Dict]) -> str:
+        """Create a Mermaid diagram representation of the flow."""
+        lines = ["graph TD"]
+        
+        # Build flow connections
+        connections = []
```

Ruff (0.12.2)

Local variable `connections` is assigned to but never used

Remove assignment to unused variable `connections`

(F841)